File: Frontend/ready2road/src/app/model/Conversazione.ts
import {Utente} from "./Utente";
import {Admin} from "./Admin";
import {Venditore} from "./Venditore";
import {Inject} from "@angular/core";
import {ConversazioneService} from "../services/conversazione.service";

export class Conversazione{
  private nome: string;
  private utente:Utente;
  private admin:Admin;
  private venditore:Venditore;

  constructor(json:any,@Inject(ConversazioneService) private service: ConversazioneService) {
    this.nome=json.nome;
    this.utente=json.utente;
    this.admin=json.admin;
    this.venditore=json.venditore;
  }
  getNome= (): string =>{
    return this.nome;
  }
  getUtente= (): Utente =>{
    return this.utente;
  }
  getAdmin= (): Admin =>{
    return this.admin;
  }
  getVenditore= (): Venditore =>{
    return this.venditore;
  }
  setNome= (nome:string): void =>{
    this.nome=nome;

  }
  setUtente= (utente:Utente): void =>{
    this.utente=utente;

  }
  setAdmin= (admin:Admin): void =>{
    this.admin=admin;

  }


}

File: Ready2RoadBackend/src/main/java/Ready2road/Persistenza/Model/Conversazione.java
package Ready2road.Persistenza.Model;

import com.google.gson.JsonObject;

public class Conversazione {
    private String nome;
    private Utente utente;
    private Venditore venditore;
    private Admin admin;

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Utente getUtente() {
        return utente;
    }

    public void setUtente(Utente utente) {
        this.utente = utente;
    }

    public Venditore getVenditore() {
        return venditore;
    }

    public void setVenditore(Venditore venditore) {
        this.venditore = venditore;
    }

    public Admin getAdmin() {
        return admin;
    }

    public void setAdmin(Admin admin) {
        this.admin = admin;
    }

    public JsonObject toJson() {
        JsonObject jsonObject = new JsonObject();
        jsonObject.addProperty("nome", nome);


        return jsonObject;
    }
}
